ports/samd/machine_i2c.c

```diff
@@ -119,25 +120,34 @@ void common_i2c_irq_handler(int i2c_id) {
 
 static void machine_i2c_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     machine_i2c_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    mp_printf(print, "I2C(%u, freq=%u, scl=%u, sda=%u)",
-        self->id, self->freq, self->scl, self->sda);
+    mp_printf(print, "I2C(%u, freq=%u, scl=\"%q\", sda=\"%q\")",
+        self->id, self->freq, pin_find_by_id(self->scl)->name, pin_find_by_id(self->sda)->name);
 }
 
 mp_obj_t machine_i2c_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
     enum { ARG_id, ARG_freq, ARG_scl, ARG_sda };
     static const mp_arg_t allowed_args[] = {
-        { MP_QSTR_id, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        #if MICROPY_HW_DEFAULT_I2C_ID < 0
+        { MP_QSTR_id, MP_ARG_REQUIRED | MP_ARG_INT, {.u_int = -1} },
+        #else
+        { MP_QSTR_id, MP_ARG_INT, {.u_int = MICROPY_HW_DEFAULT_I2C_ID} },
+        #endif
         { MP_QSTR_freq, MP_ARG_INT, {.u_int = DEFAULT_I2C_FREQ} },
-        { MP_QSTR_scl, MP_ARG_REQUIRED | MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
-        { MP_QSTR_sda, MP_ARG_REQUIRED | MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+        #if defined(pin_SCL) && defined(pin_SDA)
+        { MP_QSTR_scl, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = pin_SCL} },
+        { MP_QSTR_sda, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = pin_SDA} },
+        #else
+        { MP_QSTR_scl, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+        { MP_QSTR_sda, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
```

Comment:
Do you think in this case, of no default scl/sda, these args should retain the `MP_ARG_REQUIRED` flag?

(Also, if the default ID is 0 but you do `machine.I2C(1)` then it configures instance #1 with the default pins used by instance #0.  I guess that's OK, but maybe a better way would be to only use the default pins if the default instance is selected?)

Reply:
You're right, these must be required in case of missing defaults. 
About the second question: SAMD21 allow to use two different IDs at one pin pair for I2C and UART. For example the UART ID  for RX and TX of a ItsyBitsy M0 may be 0 or 2. At SAMD51 it's similar in that the same pin pair may be used for UART and I2C, but with different IDs the pins are swapped.